binaries/cuprated/src/txpool/incoming_tx.rs

```diff
@@ -183,29 +197,36 @@ async fn handle_incoming_txs(
         // TODO: this could be a DoS, if someone spams us with txs that violate these rules?
         // Maybe we should remember these invalid txs for some time to prevent them getting repeatedly sent.
         if let Err(e) = check_tx_relay_rules(&tx, context) {
-            tracing::debug!(err = %e, tx = hex::encode(tx.tx_hash), "Tx failed relay check, skipping.");
+            if drop_relay_rule_errors {
+                tracing::debug!(err = %e, tx = hex::encode(tx.tx_hash), "Tx failed relay check, skipping.");
+                continue;
+            }
 
-            continue;
+            return Err(IncomingTxError::RelayRule(e));
         }
```

Comment:
`/send_raw_transaction` needs this to check for errors; all previous callers maintain the same behavior.